### DISCUSSION TITLE
delegatecall 调用失败

### DISCUSSION THREAD

**Discussion by superwildboy** [AmazingAng/WTF-Solidity#454] [Q&A]
##练习23 delegatecall 调用合约C失败 
function delegatecallSetVars(address _addr, uint _num) external payable{
        // delegatecall setVars()
        (bool success, bytes memory data) = _addr.delegatecall(
            abi.encodeWithSignature("setVars(uint256)", _num)
        );
        if(!success){
            num = _num;
        }
    }
![image](https://user-images.githubusercontent.com/25010669/226542659-dd249382-c009-49e0-b1f7-d32fe60c5c6b.png)

**Comment by RealWillsLi** [comment]
我试了下没报错，这样用没问题，可能你是想调用call来改变另一个合约的值，delegatecall和call是有区别的

**Comment by XdpCs** [comment]
暂时无法提供更多的信息，就先关闭了，如有需求，请补全你的问题信息，重新打开讨论。
